### PR TITLE
Fix: check SNS neuron balances if force query

### PR DESCRIPTION
**Pull request by lmuntaner**
# Motivation

There is a bug in mainnet.

If a user starts staking an SNS neuron, but loses connection after the transfer happens but before the claiming happens, the neuron isn't claimed when the user returns.

The reason was because we were checking the balances only on update calls. We lost that when we forced only query calls with `FORCE_CALL_STRATEGY`.

# Changes

* Check SNS neuron balances also if `FORCE_CALL_STRATEGY` forces only a `"query"` strategy.

# Tests

* I added the same test to claim a neuron forcing the strategy to `"query"`, the test failed. As expected with the bug.
* Then I added the fix to also check when `FORCE_CALL_STRATEGY` is `"query"` and it passed.

# Todos

- [x] Add entry to changelog (if necessary).

### CODE REVIEW

**Comment by lmuntaner** [comment]
Current dependencies on/for this PR:
* `main`
  * **PR #4415** <a href="https://app.graphite.dev/github/pr/dfinity/nns-dapp/4415?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/dfinity/nns-dapp/4415?utm_source=stack-comment).

**Comment by lmuntaner** [comment]
@dskloetd please review

File: frontend/src/lib/services/sns-neurons.services.ts [hunk 1]
```diff
@@ -108,7 +108,7 @@ export const syncSnsNeurons = async (
         certified,
       });
 
-      if (certified) {
+      if (certified || FORCE_CALL_STRATEGY === "query") {
```
Comment (dskloetd) [inline frontend/src/lib/services/sns-neurons.services.ts:111]: There is a function for this check called `isForceCallStrategy`.
On line 129 we use the equivalent for the opposite case so we might want to use it here as well.

File: frontend/src/tests/lib/services/sns-neurons.services.spec.ts [hunk 2]
```diff
@@ -257,6 +258,45 @@ describe("sns-neurons-services", () => {
       expect(spyNeuronQuery).toBeCalled();
     });
 
+    it("should claim neuron if find a subaccount without neuron for query calls if FORCE_CALL_STRATEGY is query", async () => {
+      mockedConstants.FORCE_CALL_STRATEGY = "query";
```
Comment (dskloetd) [inline frontend/src/tests/lib/services/sns-neurons.services.spec.ts:262]: Please call `resetMockedConstants` in `beforeEach`.

**Review by dskloetd — APPROVED** [review]

File: frontend/src/lib/services/sns-neurons.services.ts (hunk 1 above)
Comment (lmuntaner) [reply frontend/src/lib/services/sns-neurons.services.ts]: Thanks, done

File: frontend/src/tests/lib/services/sns-neurons.services.spec.ts (hunk 2 above)
Comment (lmuntaner) [reply frontend/src/tests/lib/services/sns-neurons.services.spec.ts:266]: Ah, I thought they were reset automatically, thanks!

**Pull request merged**